### ISSUE TITLE
Crash when loading profile.txt

### ISSUE THREAD

**Issue by gfour** [souffle-lang/souffle#1898]
The Souffle profiler crashes when loading the attached profile. The error is the following:
```
exception whilst reading profile DB: Parse error: invalid escape character '.' (46)
souffle-profile: ./include/souffle/utility/MiscUtil.h:195: void souffle::fatal(const char*, const Args& ...) [with Args = {const char*}]: Assertion `false && "fatal error; see std err"' failed.
[2]    11721 abort (core dumped)  souffle-profile test-ivy-open-programs-profile.txt
```

Looking for escaped dots in the file shows these two lines coming from strings containing escape sequences:
```
$ fgrep '\.' test-ivy-open-programs-profile.txt
      "Stats_Simple_JavaUtilVarPointsTo(?hctx,?value,?ctx,?var) :- \n   Method_DeclaringType(?method,?type),\n   \"java\\.util\\\..*\" match ?type,\n   Var_DeclaringMethod(?var,?method),\n   mainAnalysis.VarPointsTo(?hctx,?value,?ctx,?var).": {
        "Stats_Simple_JavaUtilVarPointsTo(?hctx,?value,?ctx,?var) :- \n   Method_DeclaringType(?method,?type),\n   \"java\\.util\\\..*\" match ?type,\n   Var_DeclaringMethod(?var,?method),\n   mainAnalysis.VarPointsTo(?hctx,?value,?ctx,?var).": {
```
Deleting these `\.` strings makes the problem go away and the profiler can load the file successfully.

[test-ivy-open-programs-profile.txt.gz](https://github.com/souffle-lang/souffle/files/6303614/test-ivy-open-programs-profile.txt.gz)

**Comment by gfour** [comment]
Duplicate of #1713, should be closed (I observed the bug in the 2.0.2 release, that fix was after the release so git HEAD should work).

**Comment by b-scholz** [comment]
I still want to fix few problems before going ahead with a new release.
